core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -233,25 +233,25 @@ public void applyTo(TableMetadata.Builder metadataBuilder) {
   }
 
   class SetSnapshotRef implements MetadataUpdate {
-    private final String name;
+    private final String refName;
```

Comment:
Given we updated the spec to use `ref-name`, I also changed the field name and the method name.

I can do this in a separate PR (as well as update the spec in that same PR) or I can leave the class's code as is for now. WDYT? Changing it here does admittedly make the PR's scope somewhat broad for what it touches.